src/vsr/replica.zig

```diff
@@ -2875,8 +2884,34 @@ pub fn ReplicaType(
             assert(message.header.replica != self.replica);
 
             const op = message.header.prepare_op;
+
+            const view = message.header.view;
+            const checksum = blk: {
+                if (message.header.view == 0) {
+                    break :blk message.header.prepare_checksum;
+                }
+
+                assert(message.header.prepare_checksum == 0);
+                if (self.journal.header_with_op(op)) |header| {
+                    if (header.view > view) {
```

Comment:
I'm not sure I understand why this branch is necessary. 

Is it just preventing sending a prepare to the requester which they will ignore, or is there more to it than that?

(Can you also add a (code) comment explaining why this branch is necessary?)